pkg/admission/prep_downscale.go

```diff
@@ -159,6 +159,22 @@ func prepareDownscale(ctx context.Context, logger log.Logger, ar v1.AdmissionRev
 				ar.Request.Resource.Resource, ar.Request.Name, ar.Request.Namespace, *oldReplicas, *newReplicas, foundSts.name, foundSts.lastDownscaleTime, foundSts.waitTime,
 			)
 		}
+
```

Comment:
Seems like we're fetching the statefulsets part of this rollout group several times as part of this webhook. Would it make sense to fetch all of them once and pass the objects to each function that needs to check a particular thing about them?